Library/Formula/samba4.rb

```diff
@@ -0,0 +1,107 @@
+require 'formula'
+
+class Samba < Formula
+  homepage 'http://samba.org/'
+  url 'http://www.samba.org/samba/ftp/stable/samba-4.1.14.tar.gz'
+  sha1 'db536187b4450adc610678fc8423cb1bfdc57836'
+
+  conflicts_with 'talloc', :because => 'both install `include/talloc.h`'
+  conflicts_with 'samba', :because => 'don´t install both packages'
+  
+  depends_on "pkg-config" => :build
+  depends_on "gnutls" => :recommended
+  depends_on "popt" => :recommended  
+  
+  skip_clean 'private'
+  skip_clean 'var/locks'
+
+  # Fixes
+  patch :DATA
+
+  def install
+    system "./buildtools/bin/waf configure", "--prefix=#{prefix}",
```

Comment:
The install name bug could take sometime to be fixed by samba team. Is posible we can add like "samba4" formula on "homebrew-versions" setting build dir folder same has install dir, ex: /usr/local/Cellar/samba/4.1.16, and disable clean build dir after brew install?